tools/module-textures/pngkit.py

```
@@ -0,0 +1,211 @@
+    gmax = (1 << bit_depth) - 1
+
+    out = bytearray(w * h * 4)
+    o = 0
+    for y in range(h):
+        line = lines[y * stride:(y + 1) * stride]
+        for s in _samples(line, w, channels, bit_depth):
+            if color_type == 0:  # grayscale
+                g = s[0] * 255 // gmax if bit_depth < 8 else s[0]
+                a = 0 if trns_gray is not None and s[0] == trns_gray else 255
+                r = gg = bb = g
+                out[o:o + 4] = bytes([r, gg, bb, a]); o += 4
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)